cloud/examples/cloud_statistics_example.cc

```diff
@@ -0,0 +1,125 @@
+  if (keyid == nullptr || secret == nullptr) {
+    fprintf(
+        stderr,
+        "Please set env variables "
+        "AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY with cloud credentials");
+    return -1;
+  }
+  cloud_env_options.credentials.access_key_id.assign(keyid);
+  cloud_env_options.credentials.secret_key.assign(secret);
+  cloud_env_options.region = "us-west-2";
+
+  // Connect the cloud environment instrumentation with a stats receiver.
+  cloud_env_options.cloud_statistics = CreateCloudStatistics();
+
+  // Append the user name to the bucket name in an attempt to make it
+  // globally unique. S3 bucket-namess need to be groblly unique.
+  // If you want to rerun this example, then unique user-name suffix here.
+  char* user = getenv("USER");
+  kBucketSuffix.append(user);
+
+  // Create a new AWS cloud env Status
+  CloudEnv* cenv;
+  Status s =
+      CloudEnv::NewAwsEnv(Env::Default(), kBucketSuffix, kDBPath, kBucketSuffix,
+                          kDBPath, cloud_env_options, nullptr, &cenv);
+  if (!s.ok()) {
+    fprintf(stderr, "Unable to create cloud env bucket suffix %s. %s\n",
+            kBucketSuffix.c_str(), s.ToString().c_str());
+    return -1;
+  }
+  cloud_env.reset(cenv);
+
+  // Create options and use the AWS env that we created earlier
+  Options options;
+  options.env = cloud_env.get();
+  options.create_if_missing = true;
+
+  // Connect the DB instrumentation with the cloud stats receiver.
+  // This allows the statistics collected in the cloud environment to be shown
+  // along with the database statistics in the logs during compaction operations.
+  options.statistics = cloud_env_options.cloud_statistics;
+
+  // No persistent read-cache
+  std::string persistent_cache = "";
+
+  // open DB
+  DBCloud* db;
+  s = DBCloud::Open(options, kDBPath, persistent_cache, 0, &db);
+  if (!s.ok()) {
+    fprintf(stderr, "Unable to open db at path %s with bucket %s. %s\n",
+            kDBPath.c_str(), kBucketSuffix.c_str(), s.ToString().c_str());
+    return -1;
+  }
+
+  // Put key-value
+  s = db->Put(WriteOptions(), "key1", "value");
+  assert(s.ok());
+  std::string value;
+  // get value
+  s = db->Get(ReadOptions(), "key1", &value);
+  assert(s.ok());
+  assert(value == "value");
+
+  // atomically apply a set of updates
+  {
+    WriteBatch batch;
+    batch.Delete("key1");
+    batch.Put("key2", value);
+    s = db->Write(WriteOptions(), &batch);
+  }
+
+  s = db->Get(ReadOptions(), "key1", &value);
+  assert(s.IsNotFound());
+
+  db->Get(ReadOptions(), "key2", &value);
+  assert(value == "value");
+
+  // print all values in the database
+  rocksdb::Iterator* it = db->NewIterator(rocksdb::ReadOptions());
+  for (it->SeekToFirst(); it->Valid(); it->Next()) {
+    std::cout << it->key().ToString() << ": " << it->value().ToString()
+              << std::endl;
+  }
+  delete it;
+
+  // Flush all data from main db to sst files. Release db.
+  db->Flush(FlushOptions());
```

Comment:
Can we print out the Statsitics values into stdout values? Then this example will be super clear that it is showing how Statistics work!